### PR TITLE
Fixed #30426 -- Changed X_FRAME_OPTIONS setting default to DENY.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: docs/releases/3.0.txt [hunk 1]
```diff
@@ -535,6 +535,13 @@ upload handler is used.
 ``FILE_UPLOAD_PERMISSION`` now defaults to ``0o644`` to avoid this
 inconsistency.
 
+New default value for the ``X_FRAME_OPTIONS`` setting
+-----------------------------------------------------
+
+In older versions, the :setting:`X_FRAME_OPTIONS` setting defaults to
+``SAMEORIGIN``. To make Django projects more secure by default, the setting now
+defaults to ``DENY``.
```
Comment (adamchainz) [inline docs/releases/3.0.txt]: Maybe one more sentence to spell out the implications, like:

```
If your site uses uses frames of itself, you will need to explicitly set ``X_FRAME_ORIGINS = 'SAMEORIGIN'`` for them to continue working.
```

File: docs/ref/clickjacking.txt [hunk 2]
```diff
@@ -116,6 +120,7 @@ Browsers that support ``X-Frame-Options``
 -----------------------------------------
 
 * Internet Explorer 8+
+* Edge 12+
```
Comment (felixxm) [inline docs/ref/clickjacking.txt]: I think we can add `Edge` without a version.

**Comment by felixxm** [comment]
@claudep Thanks! I pushed minor edits and changed a check message in `docs/ref/checks.txt`.

**Comment by claudep** [comment]
Thanks Mariusz!
Hopefully we can still push the `Referrer-Policy` part of PR #11735 in 3.0 before the feature freeze!

**Pull request merged**